### PR TITLE
auth lmdb: add backend commands for checking & refreshing indexes

### PR DESCRIPTION
**Pull request by Habbie**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: modules/lmdbbackend/lmdbbackend.cc [hunk 1]
```diff
@@ -2620,17 +2620,100 @@ bool LMDBBackend::getTSIGKeys(std::vector<struct TSIGKey>& keys)
...
+      // index refresh domains 12345
+      if (argv.size() < 4) {
+        return "usage: index refresh domains <ID>\n";
+      }
+
+      if (argv[2] != "domains") {
+        return "can only refresh in the domains index\n";
+      }
+
+      uint32_t id = atoi(argv[3].c_str());
```
Comment (rgacogne) [inline modules/lmdbbackend/lmdbbackend.cc:2704]:
```suggestion
      auto id = checked_stoi<uint32_t>(argv[3]);
```

**Review by rgacogne — APPROVED** [review]

File: modules/lmdbbackend/lmdbbackend.cc (hunk 1 above)
Comment (Habbie) [reply modules/lmdbbackend/lmdbbackend.cc:2704]: I didn't do that because it says `stoi`, but you're saying it should just work?

Comment (rgacogne) [reply modules/lmdbbackend/lmdbbackend.cc:2704]: Yes, it should convert the string to the corresponding `uint32_t` value, or throw an exception if the string cannot be parsed as an unsigned integer value. Unless I'm missing something and you need a different behaviour?

Comment (Habbie) [reply modules/lmdbbackend/lmdbbackend.cc:2704]: that sounds perfect. Thanks!

**Pull request merged**